pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -184,8 +184,8 @@ func newController(store model.ConfigStore, xdsUpdater model.XDSUpdater, options
 	return s
 }
 
-// convertWorkloadEntry convert wle from Config.Spec and populate the metadata labels into it.
-func convertWorkloadEntry(cfg config.Config) *networking.WorkloadEntry {
+// ConvertWorkloadEntry convert wle from Config.Spec and populate the metadata labels into it.
+func ConvertWorkloadEntry(cfg config.Config) *networking.WorkloadEntry {
```

Comment:
not a huge fan of making this public and re-converting when we could just reusing existing event handlers for WEs in the ambient index

Reply:
Existing event handlers are broken: https://github.com/istio/istio/issues/45114